server/src/main/java/org/apache/cloudstack/diagnostics/DiagnosticsServiceImpl.java

```diff
@@ -54,6 +83,35 @@
     private VirtualMachineManager vmManager;
     @Inject
     private NetworkOrchestrationService networkManager;
+    @Inject
+    private StatsCollector statsCollector;
+    @Inject
+    private DataStoreManager storeMgr;
+    @Inject
+    private BackgroundPollManager backgroundPollManager;
+    @Inject
+    private ImageStoreDetailsUtil imageStoreDetailsUtil;
+    @Inject
+    private NfsMountManager mountManager;
+
+    // This 2 settings should require a restart of the management server?
+    static final ConfigKey<Boolean> EnableGarbageCollector = new ConfigKey<>("Advanced", Boolean.class,
+            "diagnostics.data.gc.enable", "true", "enable the diagnostics data files garbage collector", false);
+    static final ConfigKey<Integer> GarbageCollectionInterval = new ConfigKey<>("Advanced", Integer.class,
+            "diagnostics.data.gc.interval", "86400", "garbage collection interval in seconds", false);
+
+    // These are easily computed properties and need not need a restart of the management server
+    static final ConfigKey<Long> DataRetrievalTimeout = new ConfigKey<>("Advanced", Long.class,
+            "diagnostics.data.retrieval.timeout", "3600", "overall data retrieval timeout in seconds", true);
+    static final ConfigKey<Long> MaximumFileAgeforGarbageCollection = new ConfigKey<>("Advanced", Long.class,
+            "diagnostics.data.max.file.age", "86400", "maximum file age for garbage collection in seconds", true);
+    static final ConfigKey<Double> DiskQuotaPercentageThreshold = new ConfigKey<>("Advanced", Double.class,
+            "diagnostics.data.disable.threshold", "0.95", "Minimum disk space percentage to initiate diagnostics file retrieval", true);
+
+    final static String COMMAND = "/bin/bash";
+    final static String PERM_KEY = "/var/cloudstack/management/.ssh/id_rsa";
```

Comment:
config keys as well, i'd say

Reply:
I disagree with the use of config keys for these, I would say a utility/helper class should be sufficient.